session/txnmanager.go

```diff
@@ -222,14 +275,33 @@ func (m *txnManager) OnStmtCommit(ctx context.Context) error {
 	if m.ctxProvider == nil {
 		return errors.New("context provider not set")
 	}
+	m.recordEvent("stmt commit")
 	return m.ctxProvider.OnStmtCommit(ctx)
 }
 
+func (m *txnManager) recordEvent(eventName string) {
+	if m.events == nil {
+		m.resetEvents()
+	}
+	m.events = append(m.events, event{event: eventName, duration: time.Since(m.lastInstant)})
+	m.lastInstant = time.Now()
+}
+
+func (m *txnManager) resetEvents() {
+	if m.events == nil {
+		m.events = make([]event, 0, 10)
+	} else {
+		m.events = m.events[:0]
+	}
+	m.enterTxnInstant = m.lastInstant
```

Comment:
Why set `enterTxnInstant` as `lastInstant`? Should it be set as `time.Now()` on entering new txn?